### PR TITLE
DBZ-37 Added support for MySQL GTIDs

### PR DESCRIPTION
**Pull request by rhauch**
The BinlogClient library is upgraded from 0.2.4 to 0.3.1, although there were no issues with just this change.

Also, because this BinlogClient library already has support for GTIDs, this PR changes the connector to makes use of that support by adding the GTIDs from the server to the offsets created by the connector and used upon restarts.

All unit and integration tests pass with this change, although only with a MySQL server instance that does not have GTIDs enabled. ~~This PR has yet to be tested with a MySQL server instance that has GTIDs enabled, and therefore is not yet ready to be merged.~~

**UPDATE: This pull request is complete and ready to be reviewed and merged.**

### CODE REVIEW

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SourceInfo.java [hunk 1]
```diff
@@ -34,9 +37,9 @@
  * 
  * <pre>
  * {
- *     "file" = "mysql-bin.000003",
- *     "pos" = 105586,
- *     "row" = 0
+ *         "file" = "mysql-bin.000003",
```
Comment (criccomini) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SourceInfo.java]: Should `gtid` be included here?

Comment (rhauch) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SourceInfo.java]: Yes, and I've corrected this in the 2nd commit.

**Comment by criccomini** [comment]
LGTM

**Comment by rhauch** [comment]
DatabaseHistory stores the DDL changes with the offset describing the position in the source where those DDL statements were found. When a connector restarts at a specific offset (supplied by Kafka Connect), connectors such as the MySQL connector reconstruct the database schemas by having DatabaseHistory load the history starting from the beginning and stopping at (or just before) the connector's starting offset. This change allows connectors to supply a custom comparison function.

To support GTIDs, the MySQL connector needed to store additional information in the offsets. This means the logic needed to compare offsets with and without GTIDs is non-trivial and unique to the MySQL connector. This commit adds a custom comparison function for offsets.

Per [MySQL documentation](https://dev.mysql.com/doc/refman/5.7/en/replication-gtids-failover.html), slaves are always expected to start with the same set of GTIDs as the master, so no matter which the MySQL connector follows it should always have the complete set of GTIDs seen by that server. Therefore:

* Two offsets with GTID sets can be compared using only the GTID sets. 
* Any offset with a GTID set is always assumed to be newer than an offset without, since it is assumed once GTIDs are enabled they will remain enabled. (Otherwise, the connector likely needs to be restarted with a snapshot and tied to a specific master or slave with no failover.)
* Two offsets without GTIDs are compared using the binlog coordinates (filename, position, and row number).
* An offsets that is identical to another except for being in snapshot mode is considered earlier than without the snapshot. This is because snapshot mode begins by recording the position of the snapshot, and once complete the offset is recorded without the snapshot flag.

All integration tests pass with a MySQL that does not use GTIDs, but they still need to be run against MySQL with GTIDs enabled. Once that is done, this enhancement should be completed.

**Comment by rhauch** [comment]
The most recent commit adds support for running the existing integration tests against a MySQL instance with GTIDs enabled by simply using `-Pgtid-mysql` or `-Dgtid-mysql` in the Maven build. It adds a Maven profile to the MySQL connector component that uses a new variation on the Docker images already used.

Also, when the MySQL connector starts up, it now queries the MySQL server to detect whether GTIDs are enabled, and if they are it will also verify that any GTID sets from the most recently recorded offset are still available in the MySQL server (similarly to how it was already doing this for binlog filenames). If the server does not have the correct coordinates/GTIDs, the connector fails with a useful error message. This commit also tests and adjusts the `GtidSet` class added earlier in this pull request to better deal with comparisons of GTID sets for proper ordering, including when adjacent intervals could be but are not combined in MySQL's string description of a GTID set.

Finally, the commit changes the connector to output MySQL's timestamp for each event using _second_ precision rather than artificially in _millisecond_ precision. To clarify the different, this change renames the field in the event's `source` structure that records the MySQL timestamp from `ts` to `ts_sec`. Similarly, the envelope's field that records the time that the connector processed each record was renamed from `ts` to `ts_ms`.

All unit and integration tests pass with the default profile and with the new GTID-enabled profile (that must be run manually).

**Pull request merged**